src/components/display/fewlines/AddIdentityInputForm.tsx

```diff
@@ -24,13 +29,20 @@ export const AddIdentityInputForm: React.FC<AddIdentityInputFormProps> = ({
         method="post"
         onSubmit={async (event) => {
           event.preventDefault();
-          return addIdentity(identity)
-            .then(() => {
-              router && router.push(`/account/logins/${type}/validation`);
-            })
-            .catch((error: Error) => {
-              throw error;
-            });
+
+          const body = {
+            callbackUrl: "/",
+            identityInput: identity,
+          };
+
+          const eventId = await fetchJson(
+            "/api/auth-connect/send-identity-validation-code",
+            HttpVerbs.POST,
+            body,
+          ).then((data) => data.json());
+
+          router &&
+            router.push(`/account/logins/${type}/validation/${eventId.data}`);
         }}
```

Comment:
Should that be in a display component?

Reply:
I put that here because I thought previously that we were getting rid of the `RenderProps` usage.
I will extract that into its business component 👍.

Reply:
Now that I think about, I should make a task and do that somewhere else, as I also need to refacto the `AddNewIdentity` business component. What do you think?

Reply:
As long as it doesn't stay too long, I'm ok with it 😁